exact_multiple_string_matching/fast_practical_multi_string_matching.py

```diff
@@ -0,0 +1,151 @@
+class AhoCorasickNode:
+    def __init__(self, depth=0):
+        self.edges = {}
+        self.final = []
+        self.fail = None
+        self.depth = depth
+
+class AhoCorasickMachine:
+    def __init__(self, patterns):
+        self.root = None
+        self._create_aho_corasick_machine(patterns)
+
+    def _create_trie(self, patterns):
+        self.root = AhoCorasickNode()
+        for pattern in patterns:
+            current_node = self.root
+            for char in pattern:
+                new_node = AhoCorasickNode(current_node.depth + 1)
+                current_node = current_node.edges.setdefault(char, new_node)
+            current_node.final.append(pattern)
+
+    def _create_aho_corasick_machine(self, patterns):
+        self._create_trie(patterns)
+        queue = []
```

Comment:
Write "queue = self.root.edges.values()" and remove append below. Please also consider using queue.Queue